### PR TITLE
Add oci_adm_knowledge_base and oci_adm_vulnerability_audit tables

### PR DESCRIPTION
**Pull request by misraved**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```sql
> select
  id,
  display_name,
  compartment_id,
  tenant_id,
  lifecycle_state as state
from
  oci_adm_knowledge_base;
[
 {
  "compartment_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr11111oz4p4h2q37cyljaq",
  "display_name": "knowledge-base-1",
  "id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaajqzknbx2ehm2yeoo4auqbie2azh2rqzrpmjdpcdut4rq",
  "state": "DELETED",
  "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
 },
 {
  "compartment_id": "ocid1.compartment.oc1..aaaaaaaapohc2v1111111111tfxrx2mhhcjfmvqexdqj754s5ojxbi3yrydq",
  "display_name": "knowledge-base-test-sp-1",
  "id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaagtocpslyuqgkpmfosseve11111sccamyb3qf774qsrrq",
  "state": "ACTIVE",
  "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
 }
]
```

```sql
> select
  id,
  display_name,
  compartment_id,
  tenant_id,
  lifecycle_state as state
from
  oci_adm_knowledge_base
where
  lifecycle_state <> 'ACTIVE';
[
 {
  "compartment_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxz111113yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
  "display_name": "knowledge-base-1",
  "id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaajqzknbx2ehm2yeoo4auq11111zh2rqzrpmjdpcdut4rq",
  "state": "DELETED",
  "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3y11111mfkr4po3hoz4p4h2q37cyljaq"
 }
]
```

```sql
> select
  id,
  display_name,
  compartment_id,
  tenant_id,
  lifecycle_state as state
from
  oci_adm_knowledge_base
where
  time_created >= now() - interval '30' day;
[
 {
  "compartment_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecx11111i3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
  "display_name": "knowledge-base-1",
  "id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaajqzknbx2ehm2yeoo4a111112azh2rqzrpmjdpcdut4rq",
  "state": "DELETED",
  "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjn11111r4po3hoz4p4h2q37cyljaq"
 },
 {
  "compartment_id": "ocid1.compartment.oc1..aaaaaaaapohc2vvmuywosa6ltf11111hhcjfmvqexdqj754s5ojxbi3yrydq",
  "display_name": "knowledge-base-test-sp-1",
  "id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaagtocpslyuqgkpmfossevee7qeasccam11111774qsrrq",
  "state": "ACTIVE",
  "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjj11111kr4po3hoz4p4h2q37cyljaq"
 }
]
```

```sql
> select
  id,
  display_name,
  compartment_id,
  tenant_id,
  lifecycle_state as state
from
  oci_adm_knowledge_base
where
  time_updated < now() - interval '90' day;
null
```

```sql
> select
  id,
  knowledge_base_id,
  vulnerabilities,
  max_observed_cvss_v2_score,
  max_observed_cvss_v3_score,
  max_observed_cvss_v2_score_with_ignored,
  max_observed_cvss_v3_score_with_ignored,
  vulnerable_artifacts_count_with_ignored,
  vulnerable_artifacts_count,
  build_type,
  display_name,
  is_success,
  lifecycle_state as state
from
  oci_adm_vulnerability_audit;
[
 {
  "build_type": "MAVEN",
  "display_name": "admvulnerabilityaudit20230607072714",
  "id": "ocid1.admvulnerabilityaudit.oc1.ap-mumbai-1.amaaaaaa6igdexaay4nm5eqwxbfql4v2yr3ltgo711111psj2spzmdg7qyya",
  "is_success": true,
  "knowledge_base_id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaagtocpslyuqgkpmfossevee7qe11111myb3qf774qsrrq",
  "max_observed_cvss_v2_score": null,
  "max_observed_cvss_v2_score_with_ignored": null,
  "max_observed_cvss_v3_score": null,
  "max_observed_cvss_v3_score_with_ignored": null,
  "state": "ACTIVE",
  "vulnerabilities": [],
  "vulnerable_artifacts_count": 0,
  "vulnerable_artifacts_count_with_ignored": 0
 }
]
```

```sql
> select
  id,
  knowledge_base_id,
  vulnerabilities,
  max_observed_cvss_v2_score,
  max_observed_cvss_v3_score,
  max_observed_cvss_v2_score_with_ignored,
  max_observed_cvss_v3_score_with_ignored,
  vulnerable_artifacts_count_with_ignored,
  vulnerable_artifacts_count,
  build_type,
  display_name,
  is_success,
  lifecycle_state as state
from
  oci_adm_vulnerability_audit
where
  lifecycle_state <> 'ACTIVE';
null
```

```sql
> select
  id,
  knowledge_base_id,
  vulnerabilities,
  max_observed_cvss_v2_score,
  max_observed_cvss_v3_score,
  max_observed_cvss_v2_score_with_ignored,
  max_observed_cvss_v3_score_with_ignored,
  vulnerable_artifacts_count_with_ignored,
  vulnerable_artifacts_count,
  build_type,
  display_name,
  is_success,
  lifecycle_state as state
from
  oci_adm_vulnerability_audit
where
  time_created >= now() - interval '30' day;
[
 {
  "build_type": "MAVEN",
  "display_name": "admvulnerabilityaudit20230607072714",
  "id": "ocid1.admvulnerabilityaudit.oc1.ap-mumbai-1.amaaaaaa6igdexaay4nm5eqwxbfql4v2yr3ltgo7711111sj2spzmdg7qyya",
  "is_success": true,
  "knowledge_base_id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaagtocpslyuqgkpmfossevee7qeasc111113qf774qsrrq",
  "max_observed_cvss_v2_score": null,
  "max_observed_cvss_v2_score_with_ignored": null,
  "max_observed_cvss_v3_score": null,
  "max_observed_cvss_v3_score_with_ignored": null,
  "state": "ACTIVE",
  "vulnerabilities": [],
  "vulnerable_artifacts_count": 0,
  "vulnerable_artifacts_count_with_ignored": 0
 }
]
```

```sql
> select
  id,
  knowledge_base_id,
  vulnerabilities,
  max_observed_cvss_v2_score,
  max_observed_cvss_v3_score,
  max_observed_cvss_v2_score_with_ignored,
  max_observed_cvss_v3_score_with_ignored,
  vulnerable_artifacts_count_with_ignored,
  vulnerable_artifacts_count,
  build_type,
  display_name,
  is_success,
  lifecycle_state as state
from
  oci_adm_vulnerability_audit
where
  is_success;
[
 {
  "build_type": "MAVEN",
  "display_name": "admvulnerabilityaudit20230607072714",
  "id": "ocid1.admvulnerabilityaudit.oc1.ap-mumbai-1.amaaaaaa6igdexaay4nm5eqwxbfql411111ltgo77zgczpsj2spzmdg7qyya",
  "is_success": true,
  "knowledge_base_id": "ocid1.admknowledgebase.oc1.ap-mumbai-1.amaaaaaa6igdexaagtocpslyuqgkpmfossevee7qeasc11111111114qsrrq",
  "max_observed_cvss_v2_score": null,
  "max_observed_cvss_v2_score_with_ignored": null,
  "max_observed_cvss_v3_score": null,
  "max_observed_cvss_v3_score_with_ignored": null,
  "state": "ACTIVE",
  "vulnerabilities": [],
  "vulnerable_artifacts_count": 0,
  "vulnerable_artifacts_count_with_ignored": 0
 }
]
```

```sql
> select
  id,
  knowledge_base_id,
  vulnerabilities,
  source ->> 'type' as source_type,
  vulnerable_artifacts_count,
  build_type,
  display_name,
  is_success,
  lifecycle_state as state
from
  oci_adm_vulnerability_audit
where
  source ->> 'type' = 'UNKNOWN';
[
 {
  "build_type": "MAVEN",
  "display_name": "admvulnerabilityaudit20230607072714",
  "id": "ocid1.admvulnerabilityaudit.oc1.ap-mumbai-1.amaaaaaa6igdexaay4nm5eqwxbfql4v2yr3ltgo711111psj2spzmdg7qyya",
  "is_success": true,
  "knowledge_base_id": "ocid1.admknowl11111ase.oc1.ap-mumbai-1.amaaaaaa6igdexaagtocpslyuqgkpmfossevee7qeasccamyb3qf774qsrrq",
  "source_type": "UNKNOWN",
  "state": "ACTIVE",
  "vulnerabilities": [],
  "vulnerable_artifacts_count": 0
 }
]
```

```sql
> select
  id,
  knowledge_base_id,
  vulnerabilities,
  max_observed_cvss_v2_score,
  max_observed_cvss_v3_score,
  max_observed_cvss_v2_score_with_ignored,
  max_observed_cvss_v3_score_with_ignored,
  vulnerable_artifacts_count_with_ignored,
  vulnerable_artifacts_count,
  build_type,
  display_name,
  is_success,
  lifecycle_state as state
from
  oci_adm_vulnerability_audit
where
  time_created >= now() - interval '5 minutes'
  and vulnerabilities is not null;
null
```
</details>

### CODE REVIEW

**Comment by misraved** [comment]
Reference - https://github.com/turbot/steampipe-plugin-oci/pull/442

File: oci/table_ oci_adm_knowledge_base.go [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+		if tags == nil {
+			tags = map[string]interface{}{}
+		}
+		for _, v := range definedTags {
+			for key, value := range v {
+				tags[key] = value
+			}
+
+		}
+	}
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_knowledge_base.go:254]: add for the system tags

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current lifecycle state of the Vulnerability Audit.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "max_observed_cvss_v2_score",
+				Description: "Maximum Common Vulnerability Scoring System Version 2 score observed for non-ignored vulnerable Application Dependencies.",
+				Type:        proto.ColumnType_STRING,
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:80]:
```suggestion
				Type:        proto.ColumnType_DOUBLE,
```

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+			},
+			{
+				Name:        "max_observed_cvss_v3_score",
+				Description: "Maximum Common Vulnerability Scoring System Version 3 score observed for non-ignored vulnerable Application Dependencies.",
+				Type:        proto.ColumnType_STRING,
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:85]:
```suggestion
				Type:        proto.ColumnType_DOUBLE,
```

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+			},
+			{
+				Name:        "max_observed_cvss_v2_score_with_ignored",
+				Description: "Maximum Common Vulnerability Scoring System Version 2 score observed for vulnerable Application Dependencies including ignored ones.",
+				Type:        proto.ColumnType_STRING,
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:90]:
```suggestion
				Type:        proto.ColumnType_DOUBLE,
```

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+			},
+			{
+				Name:        "max_observed_cvss_v3_score_with_ignored",
+				Description: "Maximum Common Vulnerability Scoring System Version 3 score observed for vulnerable Application Dependencies including ignored ones.",
+				Type:        proto.ColumnType_STRING,
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:95]:
```suggestion
				Type:        proto.ColumnType_DOUBLE,
```

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+			},
+			{
+				Name:        "vulnerable_artifacts_count",
+				Description: "Count of non-ignored vulnerable Application Dependencies.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "build_type",
+				Description: "The type of the build tool.",
+				Type:        proto.ColumnType_STRING,
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:110]:
```suggestion
				Type:        proto.ColumnType_JSON,
```

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+		if tags == nil {
+			tags = map[string]interface{}{}
+		}
+		for _, v := range definedTags {
+			for key, value := range v {
+				tags[key] = value
+			}
+
+		}
+	}
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:310]: please add for `system_tags`

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+	response, err := session.ApplicationDependencyManagementClient.GetVulnerabilityAudit(ctx, request)
+	if err != nil {
+		logger.Error("oci_adm_vulnerability_audit.getAdmVulnerabilityAudit", "api_error", err)
+		return nil, err
+	}
+	return response.VulnerabilityAudit, nil
+}
+
+//// TRANSFORM FUNCTION
+
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:278]:
```suggestion

// Priority order for tags
// 1. System Tags
// 2. Defined Tags
// 3. Free-form tags

```

File: oci/table_ oci_adm_vulnerability_audit.go [hunk 9]
```diff
@@ -0,0 +1,339 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+func tableAdmVulnerabilityAudit(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:             "oci_adm_vulnerability_audit",
+		Description:      "OCI Vulnerability Audit",
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_vulnerability_audit.go:19]:
```suggestion
		Description:      "OCI Application Dependency Management Vulnerability Audit",
```

File: docs/tables/oci_adm_knowledge_base.md
Comment (madhushreeray30) [inline docs/tables/oci_adm_knowledge_base.md:1]: please add more relevant example queries.

File: docs/tables/oci_adm_vulnerability_audit.md
Comment (madhushreeray30) [inline docs/tables/oci_adm_vulnerability_audit.md:1]: please add more relevant example queries.

File: oci/table_ oci_adm_knowledge_base.go [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+func tableAdmKnowledgeBase(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:             "oci_adm_knowledge_base",
+		Description:      "OCI Knowledge Base",
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_knowledge_base.go:19]:
```suggestion
		Description:      "OCI Application Dependency Management Knowledge Base",
```

File: oci/table_ oci_adm_knowledge_base.go [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+			},
+			{
+				Name:        "display_name",
+				Description: "The name of the Knowledge Base.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lifecycle_state",
+				Description: "The current lifecycle state of the Knowledge Base.",
+				Type:        proto.ColumnType_STRING,
```
Comment (madhushreeray30) [inline oci/table_ oci_adm_knowledge_base.go:65]:
```suggestion
				Type:        proto.ColumnType_JSON,
```

**Pull request merged**